pkg/controller/compliancescan/compliancescan_controller_test.go

```diff
@@ -384,23 +400,52 @@ var _ = Describe("Testing compliancescan controller phases", func() {
 				// Set state to DONE
 				compliancescaninstance.Status.Phase = compv1alpha1.PhaseDone
 				compliancescaninstance.Spec.Debug = true
-				reconciler.Client.Status().Update(context.TODO(), compliancescaninstance)
+				err := reconciler.Client.Status().Update(context.TODO(), compliancescaninstance)
+				Expect(err).To(BeNil())
 			})
-			It("Should return success & not clean up the scan pods", func() {
+			It("Should return success & not clean up the scan pods or secrets", func() {
 				result, err := reconciler.phaseDoneHandler(handler, compliancescaninstance, logger, dontDelete)
+				Expect(err).To(BeNil())
 				Expect(result).ToNot(BeNil())
 
 				var pods corev1.PodList
-				reconciler.Client.List(context.TODO(), &pods)
+				err = reconciler.Client.List(context.TODO(), &pods)
 				Expect(err).To(BeNil())
 				Expect(pods.Items).ToNot(BeEmpty())
 
-				// but other resources should be preserved
 				var secrets corev1.SecretList
-				reconciler.Client.List(context.TODO(), &secrets)
+				err = reconciler.Client.List(context.TODO(), &secrets)
 				Expect(err).To(BeNil())
 				Expect(secrets.Items).ToNot(BeEmpty())
 			})
 		})
+		Context("with delete flag on but debug on as well", func() {
+			BeforeEach(func() {
+				// Create the pods for the test
+				createFakeScanPods(reconciler, compliancescaninstance.Name, nodeinstance1.Name, nodeinstance2.Name)
+				createFakeRsSecret(reconciler, compliancescaninstance.Name)
+
+				// Set state to DONE
+				compliancescaninstance.Status.Phase = compv1alpha1.PhaseDone
+				compliancescaninstance.Spec.Debug = true
+				err := reconciler.Client.Status().Update(context.TODO(), compliancescaninstance)
+				Expect(err).To(BeNil())
+			})
+			It("Should return success & not clean up the scan pods", func() {
+				result, err := reconciler.phaseDoneHandler(handler, compliancescaninstance, logger, doDelete)
+				Expect(err).To(BeNil())
+				Expect(result).ToNot(BeNil())
+
+				var pods corev1.PodList
+				err = reconciler.Client.List(context.TODO(), &pods)
+				Expect(err).To(BeNil())
+				Expect(pods.Items).To(BeEmpty())
```

Comment:
Is this only empty because this is a unit test?

Reply:
ah, no, sorry and I see now how the test was confusing. There are two tests, one calls the done handler with `debug` && `delete=false` which is used on scan re-runs in which case we want the pods and the other resources to stick around. The other is called with `debug && delete=false` which is called when the scan is deleted:
```
569 func (r *ReconcileComplianceScan) scanDeleteHandler(instance *compv1alpha1.ComplianceScan, logger logr.Logger) (reconcile.Result, error) {                                                                                                                                                                                
570 »·······if common.ContainsFinalizer(instance.ObjectMeta.Finalizers, compv1alpha1.ScanFinalizer) {                                                                                                                                                                                                                         
571 »·······»·······logger.Info("The scan is being deleted")                                                                                                                                                                                                                                                                  
572 »·······»·······scanToBeDeleted := instance.DeepCopy()                                                                                                                                                                                                                                                                    
573                                                                                                                                                                                                                                                                                                                           
574 »·······»·······scanTypeHandler, err := getScanTypeHandler(r, scanToBeDeleted, logger)                                                                                                                                                                                                                                    
575 »·······»·······if err != nil && !goerrors.Is(err, compv1alpha1.ErrUnkownScanType) {                                                                                                                                                                                                                                      
576 »·······»·······»·······return reconcile.Result{}, err                                                                                                                                                                                                                                                                    
577 »·······»·······}                                                                                                                                                                                                                                                                                                         
578                                                                                                                                                                                                                                                                                                                           
579 »·······»·······// Force remove rescan annotation since we're deleting the scan                                                                                                                                                                                                                                           
580 »·······»·······if scanToBeDeleted.NeedsRescan() {                                                                                                                                                                                                                                                                        
581 »·······»·······»·······delete(scanToBeDeleted.Annotations, compv1alpha1.ComplianceScanRescanAnnotation)                                                                                                                                                                                                                  
582 »·······»·······}                                                                                                                                                                                                                                                                                                         
583                                                                                                                                                                                                                                                                                                                           
584 »·······»·······// remove objects by forcing handling of phase DONE                                                                                                                                                                                                                                                       
585 »·······»·······if _, err := r.phaseDoneHandler(scanTypeHandler, scanToBeDeleted, logger, doDelete); err != nil {                                                                                                                                                                                                         
586 »·······»·······»·······// if fail to delete the external dependency here, return with error                                                                                                                                                                                                                              
587 »·······»·······»·······// so that it can be retried                                                                                                                                                                                                                                                                      
588 »·······»·······»·······return reconcile.Result{}, err                                                                                                                                                                                                                                                                    
589 »·······»·······}                                                                                                                                                                                                                                                                                                         
590                
```
in this case we want all resources to go away.